File: 2015/1/solution.py
import sys


def main():
    floor = 0
    first_negative = None
    for i, move in enumerate(sys.stdin.readline().strip()):
        floor += 1 if move == '(' else -1

        if floor < 0 and first_negative is None:
            first_negative = i + 1

    print(floor)
    print(first_negative)

if __name__ == '__main__':
    main()

File: 2015/1/solution.rs
use std::io;

fn main() {
    let mut input = String::new();
    io::stdin()
        .read_line(&mut input)
        .expect("Failed to read line");

    let mut floor = 0;
    let mut basement = 0;

    for (num, ch) in input.split("").enumerate() {
        floor += match ch {
            "(" => 1,
            ")" => -1,
            _ => 0,
        };
        if basement == 0 && floor == -1 {
            basement = num;
        }
    }

    println!("{}", floor);
    println!("{}", basement);
}
